internal/pkg/pkg.go

```diff
@@ -51,26 +55,20 @@ func New(path string) (*Pkg, error) {
 	if err != nil {
 		return nil, err
 	}
-	var relPath string
 	var absPath string
 	if filepath.IsAbs(path) {
-		// If the provided path is absolute, we find the relative path by
-		// comparing it to the current working directory.
-		relPath, err = filepath.Rel(cwd, path)
-		if err != nil {
-			return nil, err
-		}
 		absPath = filepath.Clean(path)
 	} else {
 		// If the provided path is relative, we find the absolute path by
-		// combining the current working directory with the relative path.
-		relPath = filepath.Clean(path)
+		// combining the current working directory with the path.
 		absPath = filepath.Join(cwd, path)
```

Comment:
I think we should require that the path passed in to `pkg.New` is an absolute path. It means we avoid code deep inside kpt that depends on the current working directory. With the new definition of the DisplayPath it seems like we no longer need it.
If a user provide a relative path to a command, it is the responsibility of the cobra command to translate it into an absolute path.

Reply:
What is the downside of accepting both absolute and relative paths ? I feel this makes the logic centralized. If callers should convert the path to abs path, why do we need this *Pkg abstraction ? Just for DisplayPath ?

Reply:
With this change we no longer really rely on actual cwd (other than just turning it into an absolute path), so I'm less concerned about it. Just having code that relies on the working directory in the core libraries is awkward.
If we support this, we should make sure it is included in the tests. As far as I can see, all tests pass in the absolute path. We have a test helper function for setting and reverting changes to the cwd: https://github.com/GoogleContainerTools/kpt/blob/7e85e46110072de131fd4f57b0606b8cb1d28b06/internal/testutil/testutil.go#L824

Reply:
Pkg is not just about resolving absolute path, it is an abstraction for resolving DisplayPath, UniquePath(absolute path) and also the way to pass around package struct to maintain consistency. For example Subpackages method should not return raw strings and should return Pkg objects so that we maintain consistent references all across. A caller would expect it to be flexible to standardize the path as opposed to being restrictive of accepting only the absolute path. 

I updated the tests using Chdir but had to create method again as there is circular dependency.

Reply:
Yeah, circular dependencies is a problem for the `pkg` package. Ideally we should put the tests in the `pkg_test` package to work around this, but we would have to test `filterMetaResources` through the exported functions.